src/app.component.tsx

```diff
@@ -15,6 +16,34 @@ export function AppComponent() {
   const [showUserLocation, setShowUserLocation] = useState(false)
   const [selectedRoutes, setSelectedRoutes] = useState<Set<string>>(new Set())
 
+  useEffect(() => {
+    let lang = localStorage.getItem('language')
+    if (!lang) {
+      let client = new XMLHttpRequest()
```

Comment:
No need to make a request to server to get accepted langauges.
There is an JavaScript API for this, `navigator.languages`

Note: `XMLHttpRequest` should not be used anymore. `fetch` is the new way of making requests.